proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1205 @@ If WiFi credentials are automatically generated by the IVI system, the password
-    [configManager applyConfiguration:config completionHandler:^(NSError * _Nullable error) {
-        [self showAlert:[error localizedDescription]];
-        NSLog(@"Error %@",[error debugDescription]);
-    }];
+<struct name="DeviceInfo" since="3.0">
+    ...
+    <param name="networkingCapabilities" type="MobileNetworkingCapabilities" mandatory="false" since="x.x">
+        <description>Device's available configurations around networking</description>
+    </param>
+    ...
+</struct>
 ```
-Capabilities required to use above mentioned APIs are as follows:
+
+**HMI_API** :
+
+SDL will send mobile networking capabilities in OnAppRegistered notification within `DeviceInfo` struct
+
+```xml
+
+<enum name="NetworkHost">
+    <element name="MOBILE"/>
+      <description>
+        Use Mobile device as Wifi access point
+      </description>
+    </element>
+    <element name="VEHICLE"/>
+      <description>
+        Use Vehicle as Wifi access point
+      </description>
+    </element>
+</enum>
+
+<enum name="FrequencyBand">
+    <element name="FREQUENCY_BAND_2_4_GHZ"/>
+    <element name="FREQUENCY_BAND_5_0_GHZ"/>
+    <element name="FREQUENCY_BAND_6_0_GHZ"/>
+</enum>
+
+<struct name="MobileNetworkingCapabilities">
+
+  <description>
+    Describes sending device's availability configurations around WiFi networking.
+    This includes the ability to automatically join a network or to host one by itself.
+  </description>
+
+  <param name="autoJoinWiFiSupported" type="Boolean" mandatory="false">
+    <description>
+      Defines whether Mobile application supports programmatic join to the external network
+    </description>
+  </param>
+  <param name="hostingWiFiSupported" type="Boolean" mandatory="false">
+    <description>
+      Defines whether Mobile device is capable of hosting WiFi network.
+    </description>
+  </param>
+    <param name="preferredNetworkHost" type="NetworkHost" mandatory="false">
+      <description>
+        This describes the preference of what device (vehicle or mobile) to use as a WiFi access point.
+    </description>
+    </param>
+    <param name="wifiFrequencyBandsSupported" type="FrequencyBand" array="true" minsize="1" maxsize="100" mandatory="false">
+      <description>
+        An array of frequencies supported by the device.
+      </description>
+    </param>
+</struct>
+
+<struct name="DeviceInfo">
+  ...
+  <param name="networkingCapabilities" type="MobileNetworkingCapabilities" mandatory="false">
+        <description>Describes Device networking capabilities</description>
+  </param>
+  ...
+</struct>
 
 ```
-Hotspot Configuration
+
+#### INI file update
+
+Ini file will contain supported modes for the vehicle.
+SDL should restrict RPCs not supported by the vehicle.
+
+```ini
+[Networking]
+
+; WifiModesSupported provides information to SDL if HMI can join to an external wifi network, and if it can host Wifi network. 

```

Comment:
```suggestion
; WifiModesSupported provides information to SDL Core regarding if the HMI can join to an external WiFi network, and if it can host a WiFi network. 
```